lib/spack/spack/version.py

```diff
@@ -146,6 +146,10 @@ def underscored(self):
     def dashed(self):
         return '-'.join(str(x) for x in self.version)
 
+    @property
+    def joined(self):
```

Comment:
Looks like there's no documentation on `dotted`, `underscored`, or even `up_to()`. Let me fix this.